dask/dataframe/dask_expr/_collection.py

```diff
@@ -1404,7 +1404,11 @@ def to_dask_array(
         A Dask Array
         """
         if lengths is True:
-            lengths = tuple(self.map_partitions(len, enforce_metadata=False).compute())
+            lens = self.map_partitions(len).compute()
```

Comment:
FWIW, just removing the `enforce_metadata` seems to be sufficient:

```
In [24]: s = dd.from_pandas(pd.Series([1, 2]), npartitions=1)

In [25]: tuple(s.map_partitions(len).compute())
Out[25]: (2,)
```

Reply:
I noticed that as well but wasn't sure if ignoring metadata enforcement was added for some other reason. I actually meant to keep `enforce_metadata=False` so thanks for pointing that out. Though thinking about it more I think metadata enforcement here (with `len`) is okay, so I'll update with your suggestion